File: mobile/ios/FH07Mobile/Model/ProtocolModels.swift
import Foundation

struct ImuSample: Equatable {
    let version: Int
    let sensorId: Int
    let seq: Int
    let timestampMs: Int64
    let ax: Int
    let ay: Int
    let az: Int
    let gx: Int
    let gy: Int
    let gz: Int
}

struct ActiveSensorStatus: Equatable {
    let sensorId: Int
    let seq: Int
    let ageMs: Int
}

struct NetworkStatus: Equatable {
    let version: Int
    let uptimeMs: Int64
    let activeSensorCount: Int
    let sensors: [ActiveSensorStatus]
}

File: mobile/android/app/src/main/java/com/example/seniordesignmobileapp/model/ProtocolModels.kt
package com.example.seniordesignmobileapp.model

data class ImuSample(
    val version: Int,
    val sensorId: Int,
    val seq: Int,
    val timestampMs: Long,
    val ax: Int,
    val ay: Int,
    val az: Int,
    val gx: Int,
    val gy: Int,
    val gz: Int,
)

data class ActiveSensorStatus(
    val sensorId: Int,
    val seq: Int,
    val ageMs: Int,
)

data class NetworkStatus(
    val version: Int,
    val uptimeMs: Long,
    val activeSensorCount: Int,
    val sensors: List<ActiveSensorStatus>,
)
